src/phpDocumentor/Reflection/Php/NodesFactory.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+
+namespace phpDocumentor\Reflection\Php;
+
+use phpDocumentor\Reflection\FqsenResolver;
+use PhpParser\Lexer;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor\NameResolver;
+use PhpParser\Parser;
+
+/**
+ * Factory to create a array of nodes from a provided file.
+ * This factory will use PhpParser and NodeTraverser to do the real processing.
+ */
+class NodesFactory
+{
+    /**
+     * Parser used to parse the code to nodes.
+     *
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * Containing a number of visitors to do some post processing steps on nodes.
+     *
+     * @var NodeTraverser
+     */
+    private $traverser;
+
+    /**
+     * Initializes the object.
+     */
+    public function __construct()
+    {
+        $this->parser = new Parser(new Lexer);
```

Comment:
Can you introduce a dependency for the Parser and Traverser and instead of constructing this in the Constructor do that in a static factory method? That will make it easier to test this class and for other to inject their own setup if needed